temporalio/client.py

```diff
@@ -1550,6 +1556,7 @@ def __init__(
         self._result_run_id = result_run_id
         self._first_execution_run_id = first_execution_run_id
         self._result_type = result_type
+        self._start_workflow_response = start_workflow_response
         self.__temporal_eagerly_started = False
```

Comment:
We need to make the links from the `StartWorkflow` response available to code that is in possession of a `WorkflowHandle`. This PR is currently proposing to attach the entire reponse proto to the workflow handle, but only exposed via an underscore-prefixed attribute, so that it can be used by SDK code but it is not advertised as a stable public API for now.